sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusClientBuilder.java

```diff
@@ -830,10 +842,27 @@ private boolean isRollingSessionReceiver() {
         private ReceiveMode receiveMode = ReceiveMode.PEEK_LOCK;
         private String subscriptionName;
         private String topicName;
+        private Duration maxAutoLockRenewDuration = MAX_LOCK_RENEW_DEFAULT_DURATION;
 
         private ServiceBusReceiverClientBuilder() {
         }
 
+        /**
+         * Sets the amount of time to continue auto-renewing the lock. Setting {@link Duration#ZERO} or {@code null}
+         * disables auto-renewal.
+         *
+         * @param maxAutoLockRenewDuration the amount of time to continue auto-renewing the lock. {@link Duration#ZERO}
+         * or {@code null} indicates that auto-renewal is disabled.
+         *
+         * @return The updated {@link ServiceBusReceiverClientBuilder} object.
+         * @throws IllegalArgumentException If {#code maxAutoLockRenewDuration} is negative.
+         */
+        public ServiceBusReceiverClientBuilder maxAutoLockRenewDuration(Duration maxAutoLockRenewDuration) {
+            validateAndThrow(maxAutoLockRenewDuration);
```

Comment:
This method is only used once. It can be pulled in here.

Reply:
`validateAndThrow(maxAutoLockRenewDuration)` is used in  ServiceBusSessionReceiverClientBuilder and in ServiceBusReceiverClientBuilder, two places.